go-controller/pkg/ovn/obj_retry.go

```diff
@@ -1166,6 +1166,12 @@ func (oc *Controller) WatchResource(objectsToRetry *retryObjs) *factory.Handler
 					return
 				}
 				klog.Infof("Delete event received for resource %v %s", objectsToRetry.oType, key)
+				// If object is in terminal state, we would have already deleted it during update.
+				// No reason to attempt to delete it here again.
+				if oc.isObjectInTerminalState(objectsToRetry.oType, obj) {
+					klog.Infof("Ignoring delete event for completed resource %v %s", objectsToRetry.oType, key)
```

Comment:
hmm didn't we decide to take into consideration the case where we can see a completed pod being retried for a deletion? As in:
podA is created, podA does its business, it goes to completed state. Update Handler will get the event and try to remove its resources (deleteLogicalPort), now let's say that failed mid-way we are now retrying the deletion of completed pod and then we get a deleted event here: our usual logic is to force a deletion here now immediately, or do we just want to return from here and let the retry do its thing?

Reply:
Is it important to delete this pod right away or can we wait for `iterateRetryResources` to go over it at the next iteration?

Reply:
i think its ok to wait because at this point we have already tried to delete it on a past update event and failed, so now it is following the default backoff retry delete mechanism